internal/generate/identitytests/main.go

```diff
@@ -348,6 +366,7 @@ type ResourceDatum struct {
 	ARNService                  string
 	ARNFormat                   string
 	arnAttribute                string
+	isARNFormatGlobal           arnFormatState
```

Comment:
nit - The `is` prefix implies a boolean. Perhaps use `arnFormatState` for the field struct and `IsARNFormatGlobal` for the related method?